controllers/runnerset_controller.go

```diff
@@ -164,3 +166,13 @@ func (r *RunnerSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&v1alpha1.Runner{}).
 		Complete(r)
 }
+
+// workAroundRunnerSpecValidationBug "fixes" the runner spec by setting an empty "env" list to avoid
+// non-sense validation errors due to (I believe) a bug in controller-runtime CLient
+//
+// > validation failure list:\nspec.template.spec.env in body must be of type array: \"null\""
+func workAroundRunnerSpecValidationBug(spec *v1alpha1.RunnerSpec) {
+	if spec.Env == nil {
+		spec.Env = []corev1.EnvVar{}
+	}
+}
```

Comment:
Can you try adding 'omitempty' to 'Env' field to solve this problem?
https://github.com/summerwind/actions-runner-controller/blob/b9c6f463ed015f53dc762819b8a392634fde3cf0/api/v1alpha1/runner_types.go#L34

Reply:
Thanks! That did solve my issue.